File: app/src/main/java/com/example/artikproject/layout/StartActivity.java
package com.example.artikproject.layout;

import android.os.Bundle;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;

import com.example.artikproject.R;
import com.example.artikproject.background_work.CustomBackground;
import com.example.artikproject.background_work.Starter_MainActivity;
import com.example.artikproject.background_work.datebase.DataBase_Local;
import com.example.artikproject.background_work.theme.Theme;

public class StartActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        MainActivity.sqLiteDatabase = new DataBase_Local(getApplicationContext()).getWritableDatabase();
        Theme.setting(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.start_activity);

        // Установка нового фона [и ТУТ НЕТ затемнителя] | Должно быть после setContentView
        findViewById(R.id.start_activity).setBackground(CustomBackground.getBackground(getApplicationContext()));

        ImageView loading_ico = findViewById(R.id.loading_ico);
        new Starter_MainActivity(this, loading_ico).start();
    }
}

File: app/src/main/java/ru/agpu/artikproject/presentation/layout/activity/StartActivity.kt
package ru.agpu.artikproject.presentation.layout.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import io.realm.Realm
import ru.agpu.artikproject.R
import ru.agpu.artikproject.background_work.datebase.Const.FragmentDirection.BACK_TO_GROUP
import ru.agpu.artikproject.background_work.datebase.Const.FragmentDirection.BACK_TO_WELCOME
import ru.agpu.artikproject.background_work.datebase.DataBaseSqlite
import ru.agpu.artikproject.presentation.layout.fragment.GroupFragment
import ru.agpu.artikproject.presentation.layout.fragment.WelcomeFragment
import ru.agpu.artikproject.background_work.theme.Theme

/**
 * Стартовое активити - в нем определяется что будет происходить при нажатии на кнопку назад в
 * разных фрагментах, а так же здесь хранятся различные временные данные необходимые фрагментам
 */
class StartActivity : AppCompatActivity() {
    override fun onBackPressed() {
        when (FRAGMENT) {
            BACK_TO_GROUP -> supportFragmentManager.beginTransaction().replace(
                R.id.fragment_container_view,
                GroupFragment::class.java, null
            ).commit()
            BACK_TO_WELCOME -> supportFragmentManager.beginTransaction().replace(
                R.id.fragment_container_view,
                WelcomeFragment::class.java, null
            ).commit()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        Realm.init(this)

//        // Обновить список факультетов и групп
//        Single.just(RetrofitToRealm().loadToDatabaseSemanticGroup())
//            .observeOn(AndroidSchedulers.mainThread())
//            .subscribeOn(Schedulers.io())
//            .onErrorResumeNext {
//                Log.e("RetrofitApi", it.message!!)
//                Single.just(false)
//            }
//            .subscribe()

        // Исключительно в качестве теста -------------------------------------------
        // Загрузить расписание группы с id 2365
//        Single.just(true)
//            .observeOn(Schedulers.newThread())
//            .subscribeOn(Schedulers.newThread())
//            .doOnSubscribe { RetrofitToRealm().loadRaspisForGroup(2365) }
//            .onErrorResumeNext {
//                it.printStackTrace()
//                Single.just(false)
//            }
//            .subscribe()
        // --------------------------------------------------------------------------


        DataBaseSqlite(applicationContext).init() // Подключаемся к базе данных
        Theme.setting(this) // Применяем тему к приложению
        super.onCreate(savedInstanceState)
        setContentView(R.layout.start_activity)
        }

    companion object {
        var FRAGMENT = 0 // Номер открытого фрагмента
        var SELECTED_GROUP = "" // Выбранная группа
    }
}
